scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java

```diff
@@ -0,0 +1,399 @@
+	 * schedules.
+	 * 
+	 * @param allScheduleEntitiesForApp - List of all schedules
+	 * @return
+	 */
+	private ApplicationScalingSchedules populateScheduleModel(List<ScheduleEntity> allScheduleEntitiesForApp) {
+
+		ApplicationScalingSchedules applicationScalingSchedules = new ApplicationScalingSchedules();
+		List<ScheduleEntity> specificDateSchedules = new ArrayList<ScheduleEntity>();
+		applicationScalingSchedules.setSpecific_date(specificDateSchedules);
+
+		for (ScheduleEntity scheduleEntity : allScheduleEntitiesForApp) {
+			switch (ScheduleTypeEnum.getEnum(scheduleEntity.getScheduleType())) {
+			case SPECIFIC_DATE:
+				specificDateSchedules.add(scheduleEntity);
+				break;
+			default:
+				break;
+			}
+		}
+
+		return applicationScalingSchedules;
+
+	}
+
+	/**
+	 * @param appId
+	 * @param applicationScalingSchedules
+	 */
+	public void setUpSchedules(String appId, ApplicationScalingSchedules applicationScalingSchedules) {
+		
+		// If there are schedules then only set up the missing data in the schedule entities
+		if (applicationScalingSchedules.hasSchedules()) {
+			
+			// Set up the missing data in specific date schedules list
+			setUpSchedules(appId, applicationScalingSchedules.getTimeZone(),
+					applicationScalingSchedules.getInstance_min_count(),
+					applicationScalingSchedules.getInstance_min_count(), applicationScalingSchedules.getSpecific_date(),
+					ScheduleTypeEnum.SPECIFIC_DATE);
+		}
+	}
+
+	/**
+	 * Sets the meta data in the entity like the appId, timeZone etc.
+	 * 
+	 * @param appId
+	 * @param timeZone
+	 * @param schedules
+	 * @param scheduleType
+	 */
+	private void setUpSchedules(String appId, String timeZone, Integer defaultInstanceMinCount,
+			Integer defaultInstanceMaxCount, List<ScheduleEntity> schedules, ScheduleTypeEnum scheduleType) {
+		if ( schedules != null && !schedules.isEmpty()) {
+			for (ScheduleEntity scheduleEntity : schedules) {
+				scheduleEntity.setAppId(appId);
+				scheduleEntity.setTimeZone(timeZone);
+				scheduleEntity.setDefaultInstanceMinCount(defaultInstanceMinCount);
+				scheduleEntity.setDefaultInstanceMaxCount(defaultInstanceMaxCount);
+				scheduleEntity.setScheduleType(scheduleType.getDbValue());
+			}
+		}
+	}
+
+	/**
+	 * This method does the basic data validation and calls the helper method to
+	 * do further validation.
+	 * 
+	 * @param appId
+	 * @param timeZone
+	 * @param applicationScalingSchedules
+	 */
+	public void validateSchedules(String appId, ApplicationScalingSchedules applicationScalingSchedules) {
+		logger.info("Validate schedules for application: " + appId);
+		
+		boolean isValid = true;
+		boolean isValidTimeZone = true; // Flag added since date time checks
+										// depend on the time zone
+
+		// Validate the application id
+		isValid = DataValidationHelper.isNotNull(appId);
+		if (!isValid) {
+			validationErrorResult.addFieldError(applicationScalingSchedules, "schedule.data.value.null", "app_id");
```

Comment:
The problem here is that when the appId is null, how do we refer to this thing in some meaningful way.